parser/object_parse/die_parse/item_kind.rs

```diff
@@ -0,0 +1,133 @@
+use gimli;
+use ir;
+
+use super::FallilbleOption;
+
+/// Calculate the kind of IR item to represent the code or data associated with
+/// a given debugging information entry.
+pub fn item_kind<R>(
+    die: &gimli::DebuggingInformationEntry<R, R::Offset>,
+    _debug_types: &gimli::DebugTypes<R>,
+    _compilation_unit: &gimli::CompilationUnitHeader<R, <R as gimli::Reader>::Offset>,
+) -> FallilbleOption<ir::ItemKind>
+where
+    R: gimli::Reader,
+{
+    let item_kind = match die.tag() {
+        gimli::DW_TAG_null => unimplemented!(),
```

Comment:
All the `unimplemented!()`s in here should just return `None`. It is OK if we don't support everything yet, but we need to just return errors or partial information, rather than panicking.